src/components/LeadsFilter/LeadsFilter.tsx

```diff
@@ -0,0 +1,126 @@
+'use client';
+
+import * as React from 'react';
+import { useForm, Controller } from 'react-hook-form';
+import Box from '@mui/material/Box';
+import InputLabel from '@mui/material/InputLabel';
+import OutlinedInput from '@mui/material/OutlinedInput';
+import MenuItem from '@mui/material/MenuItem';
+import FormControl from '@mui/material/FormControl';
+import Select from '@mui/material/Select';
+import { IconButton, InputAdornment, TextField } from '@mui/material';
+import Iconify from 'components/iconify/iconify';
+import { leadStatuses } from 'constants/leadStatuses';
+import { useFilterLeadsMutation, useFilterLeadsByPropertyMutation } from 'store/api/qobrixApi';
+
+type Props = {
+  selectedPropertyId?: string;
+};
+
+type FormValues = {
+  status: string;
+  leadName: string;
+};
+
+export default function LeadsFilter({
+  // selectedPropertyId = '006251f6-39d4-4944-bb2f-c411e0b8d98b',

```

Comment:
Commented code

Reply:
Vyacheslav, I left this code for Sergii to test the filter functionality